### PR TITLE
translate poplar exception to odla_status error

### PR DESCRIPTION
**Pull request by yanwei-gr**
Based on pr#399 and sdk 2.2.

### CODE REVIEW

File: ODLA/platforms/odla_popart/odla_compute.cc [hunk 1]
```diff
@@ -216,7 +217,40 @@ odla_status odla_ExecuteComputation(odla_computation comp, odla_context context,
...
+  }
+  catch (poplar::recoverable_runtime_error &e) {
+    popart::logging::info("Poplar recoverable_runtime_error exception caught");
+    auto action = e.getRecoveryAction();
+    popart::logging::info("need to take action:{}", action);
+    if (action == poplar::RecoveryAction::IPU_RESET) {
+      return RECOVERABLE_ERR;
+    }
+    else if (action == poplar::RecoveryAction::PARTITION_RESET) {
+      return PARTITION_RESET;
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/odla_compute.cc:235]: are these odla_status error code?

Comment (yanwei-gr) [reply ODLA/platforms/odla_popart/odla_compute.cc:235]: yes，based on pr#399 https://github.com/alibaba/heterogeneity-aware-lowering-and-optimization/pull/399

Comment (yanwei-gr) [reply ODLA/platforms/odla_popart/odla_compute.cc:235]: and need sdk2.2 to use these poplar exceptions.

Comment (weimingzha0) [reply ODLA/platforms/odla_popart/odla_compute.cc:235]: I see. can you change your local branch to ipu_testing_sdk2.2.0 ?

**Pull request closed without merging**